folio-spring-cql/src/main/java/org/folio/spring/cql/Cql2JpaCriteria.java

```diff
@@ -298,28 +298,40 @@ private Predicate indexNode(Path<?> field, CQLTermNode node, CqlModifiers modifi
       return buildModifiersQuery(field, modifiers, cb);
     }
 
-    var isString = String.class.equals(field.getJavaType());
     var comparator = node.getRelation().getBase().toLowerCase();
     var term = node.getTerm();
 
+    if (equalsAny(comparator, "=", "==") && (StringUtils.isBlank(term) || term.equals("''"))) {
+      return definedOrDefinedAndEmptyQuery(field, comparator, cb);
+    }
+
     return switch (comparator) {
       case "=" -> modifiers.getCqlTermFormat() == NUMBER
         ? queryBySql(field, term, comparator, cb)
-        : buildQuery(field, node, isString, comparator, cb);
-      case "adj", "all", "any", "==", NOT_EQUALS_OPERATOR -> buildQuery(field, node, isString, comparator, cb);
-      case "<", ">", "<=", ">=" -> queryBySql(field, term, comparator, cb);
+        : buildQuery(field, term, comparator, cb);
+      case "adj", "all", "any", "==", NOT_EQUALS_OPERATOR -> buildQuery(field, term, comparator, cb);
+      case "<", ">", "<=", ">=" -> queryBySql(field, node.getTerm(), comparator, cb);
       default -> throw new CQLFeatureUnsupportedException("Relation " + comparator + " not implemented yet: " + node);
     };
   }
 
-  @SuppressWarnings("unchecked")
-  private Predicate buildQuery(Path<?> field, CQLTermNode node, boolean isString, String comparator,
-                               CriteriaBuilder cb)
-    throws QueryValidationException {
+  private Predicate definedOrDefinedAndEmptyQuery(Path<?> field, String comparator, CriteriaBuilder cb) {
+    boolean isString = String.class.equals(field.getJavaType());
+    if ("==".equals(comparator) && isString) {
```

Comment:
What's the purpose of comparing to "==" if you already compared it before method call? And what was the purpose of comparing to "="?

Reply:
querying some field with "=" or "==" comparator is handled separately from all the other comparators and there is one difference in interpretation between "=" and "==": ```field=""``` means the field is defined (has any value - notNull) and ```field==""``` means the field is defined and empty, it's described also in the [page](https://github.com/folio-org/raml-module-builder?tab=readme-ov-file#cql-matching-undefined-or-empty-values) in a more detail